nativelib/src/main/resources/gc/commix/Sweeper.c

```diff
@@ -0,0 +1,482 @@
+If there is a superblock that crosses the batch boundary, it is handled in the batch where it starts.
+Sweeper finds free superblocks (i.e. range of free blocks) within its batch.
+
+If the Sweeper would immediately return the free superblocks to BlockAllocator then we couldn't allocate anything
+bigger than a batch. Therefore the free blocks at the beginning and the end of the batch are marked as `block_coalesce_me`.
+There will be coalesced into bigger blocks by `Sweeper_LazyCoalesce`. Other free superblocks CAN be immediately
+returned to BlockAllocator because their size is already fixed by other non-free blocks around them.
+
+Coalescing could be done in single pass over the heap once all the batches are swept. However, then large areas
+of free blocks wouldn't be available for allocation.
+Instead coalescing is done incrementally - until we reach a batch that has not been swept.
+Coalescing progress is tracked by `heap->sweep.coalesceDone` - coalescing was done up to this point.
+Each sweeper has cursorDone (even the lazy sweeper) to track how far have we swept.
+
+Coalescing is done incrementally - `Sweeper_LazyCoalesce` is called after each batch is swept.
+There can be no more than 1 thread coalescing at a time. If a thread notices a coalescing is in progress it
+does not interfere and does not retry until it tries to sweep another batch.
+Coalescing might take a long time and lead to longer GC pauses and inconsistent performance.
+If at all possible (i.e. SCALANATIVE_GC_THREADS > 0), we avoid running it on the mutator thread.
+
+When the coalescing reaches the end of heap sweeping is done. After sweeping `Sweeper_sweepDone` is called
+on the mutator thread. This is done to avoid synchronization in the `Heap_Grow`. It is only called from the mutator
+thread, therefore no synchronization is needed.
+
+EXAMPLE:
+SWEEP_BATCH_SIZE=3, there are 9 blocks in total and 2 threads
+? - unswept block F-free U-unavailable, C-coalesce_me, [] - superblock
+
+???|???|???
+thread 1 starts sweeping the first batch and thread 2 the second one
+U??|F??|???
+UFF|F??|???
+U[CC]|F??|???
+Thread 1 is done with its block and attempts to coalesce.
+The second batch is not done, so it can only coalesce up to item 4.
+U[FF]|FF?|???
+It starts creating a superblock, then starts sweeping batch 3.
+U[FF]|FF?|UF?
+U[FF]|FFU|UF?
+U[FF]|FFU|U[F]U
+In Batch 3 the free block in the middle gets immediately returned to BlockAllocator.
+Thread 1 is done, tries to coalesce, but there is nothing to do because batch 2 is not done yet.
+U[FF]|[CC]U|U[F]U
+Batch 2 is done, thread 2 tries to coalesce. It can coalesce the remaining 2 batches.
+U[FFFF]U|U[F]U
+The free superblock of size 4 gets returned to BlockAllocator.
+U[FFFF]U|U[F]U
+U[FFFF]UU[F]U
+All is coalesced! Sweeping is done.
+
+Note that besides coalescing `Sweeper_LazyCoalesce` also finishes the sweeping of superblocks in some cases.
+See also `block_superblock_start_me` and `LargeAllocator_Sweep`.
+*/
```

Comment:
A few changes to formatting:
1. Align doc comments on 80 character line limit
2. Use `//` for non-doc comment style comments.
3. Indent examples by 4 spaces and blank line before and after them.
4. Make sure that comment is up-to-date with latest implementation.